File: android/app/src/main/java/com/example/unio/presentation/auth/AuthContract.kt
package com.example.unio.presentation.auth

import com.example.unio.domain.model.User

data class AuthState(
    val email: String = "",
    val password: String = "",
    val displayName: String = "",
    val isLoading: Boolean = false,
    val error: String? = null
)

sealed interface AuthIntent {
    data class UpdateEmail(val email: String) : AuthIntent
    data class UpdatePassword(val password: String) : AuthIntent
    data class UpdateDisplayName(val name: String) : AuthIntent
    data object Register : AuthIntent
    data object Login : AuthIntent
    data object ClearError : AuthIntent
}

sealed interface AuthEffect {
    data class NavigateToHome(val user: User) : AuthEffect
    data class ShowError(val message: String) : AuthEffect
}

File: ios/Unio/Unio/Presentation/Auth/AuthContract.swift
import Foundation

struct AuthState: Equatable {
    var email: String = ""
    var password: String = ""
    var displayName: String = ""
    var isLoading: Bool = false
    var error: String? = nil
}

enum AuthIntent {
    case updateEmail(String)
    case updatePassword(String)
    case updateDisplayName(String)
    case register
    case login
    case clearError
}

enum AuthEffect: Equatable {
    case navigateToHome
    case showError(String)
}
